### DISCUSSION TITLE
How to set up self-host docker api.keygen.localhost so user can connect to self-host keygen license server via Internet?

### DISCUSSION THREAD

**Discussion by chcuongdsi** [keygen-sh/community#152] [Issue]
I am hosting Keygen in Docker for testing on my local machine. For self-host Keygen, the URL api.keygen.localhost must be kept, and we use the host file in Windows to match this URL with my local IP. It works fine within the local network, and other machines in the network can access the Keygen API through the URL api.keygen.localhost, but their host files must also be edited to match api.keygen.localhost to my local machine's IP.

My question is, how can I access this URL publicly from outside my local network so our application can access the local licensing server?

For example, our software will use API URL api.eems.com. How to link api.eems.com to api.keygen.localhost. We are struggling to set up self-host keygen in Docker via Internet so clients can connect to our licensing server?

**Comment by ezekg** [comment]
> How to link api.eems.com to api.keygen.localhost

You don't. Set `KEYGEN_HOST=api.eems.com` since that's your real domain.

The `api.keygen.localhost` is just an example.

**Comment by chcuongdsi** [comment]
Thanks for your recommendation. We created api.eems.com and assigned it to our public IP, then pointed it to our local machine that hosting 
the Keygen. From outside we can see the port 3000 open that means Keygen is beeing saw from outside of local network. How can we make use of api.eems.com for outside access? We can access https://api.keygen.localhost/v1/ping successfully but we can't access https://api.eems.com/v1/ping. "Not Found" exception occurs.

And do we need to change the host file on the local machine that hosting the Keygen?

**Comment by ezekg** [comment]
No, you should not be changing the hosts file. Doing so indicates that you're not setting things up correctly. You need to set up your actual domain, not a `.localhost` domain. You do this by setting the `KEYGEN_HOST` environment variable to your own domain name.

The value `api.keygen.localhost` should not exist in any production application — it's an example value and should be replaced with your own domain name, `api.eems.com`. I hope that makes sense?

**Reply by matrix1603** [reply]
for local testing what host we need to set, currently if we are doing localhost as value of KEYGEN_HOST, its giving below error 

 "detail": "The requested endpoint was not found (check your HTTP method, Accept header, and URL path)",
 
 from postman testing on http://localhost:3000/api/v1/accounts/<account-id>/users?
